xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -1146,7 +1162,13 @@ public String toString() {
           return StructOrError.fromError(
               "Unknown cluster specifier: " + proto.getClusterSpecifierCase());
       }
-      return StructOrError.fromStruct(new RouteAction(cluster, weightedClusters));
+      long timeoutNano = TimeUnit.SECONDS.toNanos(15L);  // default 15s
+      if (proto.hasMaxGrpcTimeout()) {
+        timeoutNano = Durations.toNanos(proto.getMaxGrpcTimeout());
```

Comment:
If configured as 0, the maximum allowed timeout for gRPC requests is infinity.
https://github.com/envoyproxy/envoy/blob/c694e470dbc9a1dd24b296d117330f0e5becb2c9/api/envoy/api/v2/route/route_components.proto#L964